src/main/java/net/conjur/api/Credentials.java

```diff
@@ -13,17 +15,27 @@
 public class Credentials {
     private static final String CONJUR_AUTHN_LOGIN_PROPERTY = "CONJUR_AUTHN_LOGIN";
     private static final String CONJUR_AUTHN_API_KEY_PROPERTY = "CONJUR_AUTHN_API_KEY";
+    private static final String CONJUR_AUTHN_SERVICE_ID_PROPERTY = "CONJUR_AUTHN_SERVICE_ID";
 
     private String username;
     private String password;
+    private String serviceID;
 
     /**
      * @param username the username/login for this Conjur identity
      * @param password the password or api key for this Conjur identity
      */
     public Credentials(String username, String password) {
+        this(username, password, "authn");
+    }
+
+    public Credentials(String username, String password, String serviceID) {
         this.username = username;
         this.password = password;
+        if((serviceID != "authn") && !serviceID.startsWith("authn-iam/") && !serviceID.startsWith("authn-oidc/")) {
```

Comment:
What is the reason for limiting this here? Why could I use `authn-ldap` for example?

Reply:
You bring up a great point, I did this to assist the user in case they misspell a service ID. If I remove `CONJUR_AUTHN_SERVICE_ID` I can then remove this if statement.